### ISSUE TITLE
[ER] Useless slicing warning

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#8089]
### What it does

Inspired by this issue:
https://github.com/rust-lang/rust/pull/91625

In code like this:

```rust
fn main() {
    let s = "abc".to_string();
    let t = s.strip_prefix(&['a', 'x'][..]);
    println!("{:?}", t);
}
```

Clippy could suggest to write instead:

```rust
s.strip_prefix(&['a', 'x'])
```

Or even:

```rust
s.strip_prefix(['a', 'x'])
```

### Example

```rust
<code>
```

Could be written as:

```rust
<code>
```

**Comment by est31** [comment]
Another example where there should be a lint:

```Rust
fn foo(_v: &[u8]) {}

fn main() {
    let v = vec![1, 2, 3, 4];
    foo(&v[..]);
}
```

**Comment by giraffate** [comment]
We have the [`redundant_slicing`](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_slicing) lint, so it would be better to improve this.

**Comment by giraffate** [comment]
> Clippy could suggest to write instead:
> ```
> s.strip_prefix(&['a', 'x'])
> ```

It causes a compile error with currently stable 1.57.0, not beta and nightly.

**Comment by est31** [comment]
FTR, this doesn't trigger the lint either:

```Rust
#![forbid(clippy::redundant_slicing)]
fn foo(_v: &[u8]) {}

fn main() {
    foo(&[1, 2, 3, 4][..]);
}
```
And the version without the `[..]` compiles on stable.
